src/emu-i386/simx86/sigsegv.c

```diff
@@ -361,6 +340,24 @@ int e_emu_fault(struct sigcontext *scp)
 	return TryMemRef;
   }
 #endif
+
+  /* CONFIG_CPUSIM only: */
+  if (_trapno==0x0e) {
+	if (in_dpmi_emu) {
+		/* reflect DPMI page fault back to a DOSEMU crash or
+		   DPMI exception;
+		   vm86 faults will terminate DOSEMU via "return 0"
+		*/
+		TheCPU.err = EXCP0E_PAGE;
+		TheCPU.scp_err = _err;
+		TheCPU.cr2 = _cr2;
+		TheCPU.eip = P0 - LONG_CS;
+		fault_cnt--;
+		siglongjmp(jmp_env, 0);
```

Comment:
Will we never hit an Interp86() fault in vm86sim mode
when interpreting realmode code? If we still can, then
removing SA_ONSTACK for no-native-dpmi case won't help.
And if we can't hit it for realmode code, then how about
just having the private SIGSEGV handler for codegen-sim.c?
I am probably missing something, but to me it looks like
in a fullsim mode there will be no other SIGSEGV sources,
so the global handler can then be completely disabled.

In fact I really doubt we should fault in a fullsim mode...
If you think that will be too slow then we can of course
leave it as it is, but IMHO who needs more speed, should
not use fullsim at first place. :)